main.go

```diff
@@ -89,8 +87,7 @@ func main() {
 		log.Fatalf("[ERR] failed to shutdown server: %s", err)
 	}
 
-	// If we got this far, it was an interrupt, so don't exit cleanly
-	os.Exit(2)
+	os.Exit(0)
```

Comment:
why are we changing the exit code from `2` to `0`? wouldn't `2` be more appropriate in case you receive events corresponding to `os.Interrupt, syscall.SIGTERM` ?

The other change does make sense to me. We also don't have any `<-serverCh` in `main.go` that would wait for `close(serverCh)`